src/components/MatchPage/MatchPage.js

```diff
@@ -1,101 +1,24 @@
 import React, { useState, useEffect } from 'react';
 import { db } from '../../services/firebase';
 import MatchCard from './MatchCard';
+import { useAuth } from '../../services/AuthContext';
 import './MatchPage.css';
 
-// class MatchPage extends React.Component{
-// 	constructor() {
-// 		super();
-// 		this.state = {
-// 			matches: [],
-// 			hasMatches: undefined
-// 		};
-// 	}
-
-// 	callInOrder(first, second){
-// 		first();
-// 		second();
-// 	}
-
-// 	componentDidMount(){
-// 		//get user id 
-// 		const userId = "test";
-// 		let likes = new Set();
-// 		let likedBy = new Set();
-// 		const likesRef = db.ref('Users/'+userId+'/Likes');
-// 		const likedByRef = db.ref('Users/'+userId+'/LikedBy');
-
-
-// 		likesRef.on('value', (snapshot) => {
-// 			if(snapshot !== null)
-// 			{
-// 				for( let i in snapshot.val())
-// 				{
-// 					likes.add(snapshot.child(i).val());
-// 				}
-// 				// console.log(likes);
-// 				likedByRef.on('value', (c_snapshot) => {
-// 				if(snapshot !== null)
-// 				{
-// 					for( let i in c_snapshot.val())
-// 					{
-// 						likedBy.add(c_snapshot.child(i).val());
-// 					}
-// 					// console.log(likedBy);
-// 					let intersection = new Set([...likes].filter(x => likedBy.has(x)));
-// 					// console.log(intersection.size);
-// 					// console.log(Array.from(intersection));
-// 					if (intersection.size  === 0)
-// 					{
-// 						this.setState({matches: null, hasMatches: false});
-// 					}
-// 					else
-// 					{
-// 						this.setState({matches: Array.from(intersection), hasMatches: true});
-// 					}
-// 				}
-// 				});
-// 			}
-// 		});
-// 	}
-
-
-
-// 	render(){
-// 		var opt = '';
-// 		if(this.state.hasMatches === undefined)
-// 		{
-// 			opt ='Loading...';
-// 		}
-// 		else if(this.state.hasMatches === false)
-// 		{
-// 			opt='No matches found!';
-// 		}
-// 		else{
-// 			opt = this.state.matches.map((id) => <MatchCard key={id} user_id={id} />)
-
-// 		}
-// 		return(
-// 			<div>
-// 				<h1> Match Page </h1>
-// 				{opt}
-// 			</div>
-// 		);
-// 	}
-// }
-
 function MatchPage() {
 	const [matches, setMatches] = useState("Loading...");
+	const {uid, userData} = useAuth();
+
 	useEffect(() => {
-		const userId = "test";
+		const userId = uid;
+		const user_data = userData;
 		let likes = new Set();
 		let likedBy = new Set();
 		const userRef = db.ref('Users/'+userId);
 		userRef.on('value', (snapshot) => {
```

Comment:
here you use the authContext to get the userData. The userData should already contain all the info regarding the user's likes and likedBys. So i don't think you have to do the whole userRef.on() stuff here, you can just use the userData from the authContext. See the AuthContext.js file, under useEffect() it shows how userData is set.